### ISSUE TITLE
Support for Bootstrap v5.3

### ISSUE THREAD

**Issue by Notch-Man** [react-bootstrap/react-bootstrap#6530]
### Is your feature request related to a problem? Please describe

Bootstrap v5.3 brings quite a good change i.e, bringing color modes in which the developer can change the theme (light or dark) of a component. This feature is helpful as it prevents writing extra styles.

### Describe the solution you'd like

I really will like this feature as it makes React developers write less code.

### Additional context

Though Bootstrap v5.3 is currently in alpha (at the time of speaking), still I will really like you to add support for Bootstrap v5.3.

**Comment by jquense** [comment]
I don't think there is any reason RB wouldn't support it out of the box. Most of the changes look like CSS additions which should just work since RB doesn't alter or even include the CSS itself

**Comment by jquense** [comment]
sure but i'm still not sure waht RB would add to support that.

**Comment by Cow258** [comment]
I'm using `Bootstrap v5.3` and `React Bootstrap v2.7`.
It is working well.
We just need to add custom theme switcher.
### Original tutorial
https://getbootstrap.com/docs/5.3/customize/color-modes/#javascript

### Usage
```jsx
import { initTheme, toggleTheme } from './Theme'

initTheme()

const App = () => (
  <button onClick={() => toggleTheme()}>
    Toggle Theme
  </button>
)
```
### Source
```jsx
/** @enum {string} */
const THEME = {
  LIGHT: 'light',
  DARK: 'dark',
}

const IS_SERVER = typeof window === 'undefined'

const getDefaultTheme = () => (
  window.matchMedia('(prefers-color-scheme: dark)').matches
    ? THEME.DARK
    : THEME.LIGHT
)

const getPreferredTheme = () => {
  const storedTheme = localStorage.getItem('theme')
  if (storedTheme) return storedTheme

  const defaultTheme = getDefaultTheme()
  localStorage.setItem('theme', defaultTheme)

  return defaultTheme
}

export function setTheme(theme) {
  if (IS_SERVER) return
  document.documentElement.dataset.bsTheme = theme
  localStorage.setItem('theme', theme)
}

export function resetTheme() {
  if (IS_SERVER) return
  setTheme(getDefaultTheme())
}

export function toggleTheme() {
  if (IS_SERVER) return
  const nextTheme =
    document.documentElement.dataset.bsTheme === THEME.DARK
      ? THEME.LIGHT
      : THEME.DARK
  setTheme(nextTheme)
}

export function initTheme() {
  if (IS_SERVER) return
  setTheme(getPreferredTheme())
}

```

**Comment by deanc** [comment]
The issues here are a little more nuanced than it might appear. For example `Navbar` is deprecating `navbar-dark` classes which react-boostrap is adding. More info here: https://getbootstrap.com/docs/5.3/components/navbar/#color-schemes

**Comment by bolora** [comment]
I agree with @deanc. I would add though that it's more than nuanced. Its a paradigm shift. 5.2 dark mode required shifting classes at component level and possibly having to customize certain css. Whereas 5.3 use one convenient global setting that triggers the dark/light theme throughout using `@if $enable-dark-mode`.

I would love to see an equally stupid simple implementation in RB. A simple setting with three possible values ["light","dark","auto"]

Something as simple as
```
<ThemeProvider theme="dark">
  <App />
</ThemeProvider>
```

**Comment by EricCote** [comment]
Here is my implementation of a simple DarkModeDropdown.  Just add this DropDown to your navbar.  

To create this, I looked at the source code from the new getBootstrap.com website. I tried to stay as close as I could to the way they worked. 

On client-side, on first load, it defaults to the user's preference. It will modify the DOM to set `data-bs-theme` on `<html>`. If you change your preference on the dropdown, it will use `localStorage` to remember your preference, using the key `theme`. (Just like on getBootstrap.com) 

On client-side, on susbsequent visits, it will use your prefences that were saved on localStorage. (Just like on getBootstrap.com) 

On server-side, this component defaults to "light" mode. (getBootstrap.com has no server-side.)  

```js

import { useState, useEffect } from 'react';
import { NavDropdown } from 'react-bootstrap';

const IS_SERVER = typeof window === 'undefined';

let storedTheme = IS_SERVER ? 'light' : localStorage.getItem('theme');

const arrayOfThemes = [
  { name: 'Light', icon: '☀️' },
  { name: 'Dark', icon: '🌙' },
  { name: 'Auto', icon: '⚙️' },
  // { name: 'Blue', icon: '🟦' }, //Add your own themes!
];


export default function DarkModeMenu() {
  const [mode, setMode] = useState(getPreferredTheme());

  useEffect(() => {
    if (IS_SERVER) return;
    if (
      mode === 'auto' &&
      window.matchMedia('(prefers-color-scheme: dark)').matches
    ) {
      document.documentElement.setAttribute('data-bs-theme', 'dark');
    } else {
      document.documentElement.setAttribute('data-bs-theme', mode);
    }
  }, []);

  function getPreferredTheme() {
    if (storedTheme) {
      return storedTheme;
    }

    return window.matchMedia('(prefers-color-scheme: dark)').matches
      ? 'dark'
      : 'light';
  }

  function setPreferredTheme(theme) {
    if (
      theme === 'auto' &&
      window.matchMedia('(prefers-color-scheme: dark)').matches
    ) {
      document.documentElement.setAttribute('data-bs-theme', 'dark');
    } else {
      document.documentElement.setAttribute('data-bs-theme', theme);
    }

    localStorage.setItem('theme', theme);
    setMode(theme);
  }

  return (
    <NavDropdown
      title={
        <>
          {
            arrayOfThemes.find((theme) => theme.name.toLowerCase() === mode)
              ?.icon
          }{' '}
        </>
      }
    >
      {arrayOfThemes.map((theme) => {
        const active = mode === theme.name.toLowerCase();
        return (
          <NavDropdown.Item
            key={theme.name}
            className={active ? 'active' : ''}
            onClick={() => {
              setPreferredTheme(theme.name.toLocaleLowerCase());
            }}
          >
            {' '}
            {theme.icon} {theme.name} {active ? '✔️' : ''}
          </NavDropdown.Item>
        );
      })}
    </NavDropdown>
  );
}


```

**Comment by EricCote** [comment]
I liked @bolora 's idea of having a theme provider. It is more flexible that way.

This sample uses a few files, so I created a codeSandbox.

https://codesandbox.io/s/themeprovider-usgsko

**Comment by loganbutler** [comment]
I came across this issue and the solution is much simpler than you'd think. 

We can simply add the `data-bs-theme` attribute anywhere we'd like, and use it however we like (with state, context, etc). 

If you want per-component theme styling, you can define the same attribute deep in your code, or if you want global control, simply add the attribute to your `<body>` tag, and all of it will fall in line. Mix in some localstorage, for setting the theme variable whenever they refresh the page, and you're done!

Tested with BS 5.3 + RB 2.7, works fine! 

`document.documentElement.setAttribute('data-bs-theme', theme)`  

The problem I came across (which is the reason I am here) is because the Navbar *does not* respond to the theme switches, and you need to also define the "bg" prop on the RB Navbar component. Otherwise, adding the `data-bs-theme` attribute anywhere else works fine!

My workaround to make it work was to just pass the theme value (e.g. dark/light) to the "bg" prop on the Navbar, and now the Navbar also switches between Dark/Light mode. I don't know if it's intended to work that way, or if I'm just an idiot.

**Comment by avand** [comment]
Working on an app where I'd _love_ to support dark mode and we're just one minor version away.

Any timeline on when it would be reasonable to expect RB to use Bootstrap v5.3? Any todo list or task list I could chip away at to help us get there?

**Comment by loganbutler** [comment]
> 3. I feel that pushing 'dark' or 'light' to the `bg` attribute is a bad practice. (And you must feel the same way too, or you would not have shared your comment.)  Instead, use `bg="body"`.  This will assign the body background for the current `data-bs-theme` (So dark when in "dark" mode, blue when in "blue" mode, etc.)

@EricCote thank you for that, i was unaware. that's absolutely better than needing to pass in the dark/light variable. 

> Working on an app where I'd _love_ to support dark mode and we're just one minor version away.
> 
> Any timeline on when it would be reasonable to expect RB to use Bootstrap v5.3? Any todo list or task list I could chip away at to help us get there?

@avand you can just install bootstrap 5.3, RB supports it just fine. as stated by others earlier in this post, RB does not modify or change the core bootstrap CSS, just provides the components. i'm sure there may need some things that need updating, but overall, works just fine if you just upgrade the bootstrap version. i have personally not noticed any problems.

**Comment by marcsoler** [comment]
Bootstrap 5.3 is now stable https://github.com/twbs/bootstrap/releases/tag/v5.3.0

**Comment by innegatives** [comment]
When is this coming to RB?

**Comment by kyletsang** [comment]
2.8.0 has been published with fixes for 5.3 and the docs have been updated.  Color modes is toggled by setting `data-bs-theme="dark"`.  See upstream docs for more details:  https://getbootstrap.com/docs/5.3/customize/color-modes/

**Comment by avand** [comment]
Oh, hell yeah! Best update I've received all week.
